source/grid/grid_in.cc

```diff
@@ -835,7 +835,31 @@ GridIn<dim, spacedim>::read_unv(std::istream &in)
             n_entities;
 
           AssertThrow(in.fail() == false, ExcIO());
-          in >> id;
+          // Ocassionally we encounter IDs which are not integers - we don't
+          // support that case since there is no sane way for us to determine
+          // integer IDs from, e.g., strings.
+          std::string line;
+          // get the newline
+          std::getline(in, line);
+          // get the next line
+          std::getline(in, line);
```

Comment:
Did you mean to call `getline` twice? If so please add a comment why you discard the first line